service/systemd/service.go

```diff
@@ -69,27 +68,23 @@ type Service struct {
 	FallBackDirName string
 	Script          []byte
 
+	fileOps FileOps
 	newDBus DBusAPIFactory
 }
 
 // NewServiceWithDefaults returns a new systemd service reference populated
 // with sensible defaults.
 func NewServiceWithDefaults(name string, conf common.Conf) (*Service, error) {
-	svc, err := NewService(name, conf, EtcSystemdDir, NewDBusAPI, renderer.Join(paths.NixDataDir, "init"))
+	svc, err := NewService(name, conf, EtcSystemdDir, NewDBusAPI, fileOps{}, renderer.Join(paths.NixDataDir, "init"))
 	return svc, errors.Trace(err)
 }
 
 // NewService returns a new reference to an object that implements the Service
 // interface for systemd.
 func NewService(
-	name string, conf common.Conf, dataDir string, newDBus DBusAPIFactory, fallBackDirName string,
+	name string, conf common.Conf, dataDir string, newDBus DBusAPIFactory, fileOps FileOps, fallBackDirName string,
 ) (*Service, error) {
 	confName := name + ".service"
-	var volName string
-	if conf.ExecStart != "" {
-		volName = renderer.VolumeName(common.Unquote(strings.Fields(conf.ExecStart)[0]))
```

Comment:
`VolumeName` always returns "" for *nix, and we know that systemd is always *nix, so this is not required.